build/akri-dotnet-containers.mk

```diff
@@ -0,0 +1,102 @@
+define add_onvif_target
+  $(info compiling akri-$(1)) # logged when doing $(call ...)
+  $$(info evaluating akri-$(1)) # logged when doing $(eval $(call ...))
+
+  akri-$(1): akri-build akri-docker-$(1)
+  akri-docker-$(1): $(1)-build $(1)-docker-per-arch $(1)-docker-multi-arch-create $(1)-docker-multi-arch-push
+  $(1)-build: $(1)-build-amd64 $(1)-build-arm32 $(1)-build-arm64
+  $(1)-docker-per-arch: $(1)-docker-per-arch-amd64 $(1)-docker-per-arch-arm32 $(1)-docker-per-arch-arm64
+
+  $(1)-build-amd64:
+  ifeq (1, ${BUILD_AMD64})
+	  docker build $(CACHE_OPTION) -f $(DOCKERFILE_DIR)/Dockerfile.$(2) . -t $(PREFIX)/$(2):$(LABEL_PREFIX)-$(AMD64_SUFFIX) --build-arg OUTPUT_PLATFORM_TAG=$(USE_OPENCV_BASE_VERSION)-$(AMD64_SUFFIX) --build-arg DOTNET_PUBLISH_RUNTIME=linux-x64
+  endif
+  $(1)-build-arm32:
+  ifeq (1, ${BUILD_ARM32})
+	  docker build $(CACHE_OPTION) -f $(DOCKERFILE_DIR)/Dockerfile.$(2) . -t $(PREFIX)/$(2):$(LABEL_PREFIX)-$(ARM32V7_SUFFIX) --build-arg OUTPUT_PLATFORM_TAG=$(USE_OPENCV_BASE_VERSION)-$(ARM32V7_SUFFIX) --build-arg DOTNET_PUBLISH_RUNTIME=linux-arm
+  endif
+  $(1)-build-arm64:
+  ifeq (1, ${BUILD_ARM64})
+	  docker build $(CACHE_OPTION) -f $(DOCKERFILE_DIR)/Dockerfile.$(2) . -t $(PREFIX)/$(2):$(LABEL_PREFIX)-$(ARM64V8_SUFFIX) --build-arg OUTPUT_PLATFORM_TAG=$(USE_OPENCV_BASE_VERSION)-$(ARM64V8_SUFFIX) --build-arg DOTNET_PUBLISH_RUNTIME=linux-arm64
+  endif
+
+  $(1)-docker-per-arch-amd64:
+  ifeq (1, ${BUILD_AMD64})
+	  docker push $(PREFIX)/$(2):$(LABEL_PREFIX)-$(AMD64_SUFFIX)
+  endif
+  $(1)-docker-per-arch-arm32:
+  ifeq (1, ${BUILD_ARM32})
+	  docker push $(PREFIX)/$(2):$(LABEL_PREFIX)-$(ARM32V7_SUFFIX)
+  endif
+  $(1)-docker-per-arch-arm64:
+  ifeq (1, ${BUILD_ARM64})
+	  docker push $(PREFIX)/$(2):$(LABEL_PREFIX)-$(ARM64V8_SUFFIX)
+  endif
+
+  $(1)-docker-multi-arch-create:
+  ifeq (1, ${BUILD_AMD64})
+	  $(ENABLE_DOCKER_MANIFEST) docker manifest create --amend $(PREFIX)/$(2):$(LABEL_PREFIX) $(PREFIX)/$(1):$(LABEL_PREFIX)-$(AMD64_SUFFIX)
+  endif
+  ifeq (1, ${BUILD_ARM32})
+	  $(ENABLE_DOCKER_MANIFEST) docker manifest create --amend $(PREFIX)/$(2):$(LABEL_PREFIX) $(PREFIX)/$(1):$(LABEL_PREFIX)-$(ARM32V7_SUFFIX)
+  endif
+  ifeq (1, ${BUILD_ARM64})
+	  $(ENABLE_DOCKER_MANIFEST) docker manifest create --amend $(PREFIX)/$(2):$(LABEL_PREFIX) $(PREFIX)/$(1):$(LABEL_PREFIX)-$(ARM64V8_SUFFIX)
+  endif
+
```

Comment:
Suggestion - Looks like a lot of the code here and the rest of the file can be further simplified by using different parameters for each of the build architectures initially then calling the docker commands once.

Reply:
i don't know how to do that with a makefile.  if someone does, i'm happy to learn.

Reply:
Haven't tried this, but would this work?

https://www.gnu.org/software/make/manual/html_node/Eval-Function.html

Reply:
(Specifically the combination with the foreach construct.)

Reply:
that's the mechanism i'm using already.  i don't know how to make a nested define and get it working.

Reply:
My bad, for some reason I only saw one iteration. Second option would be to use CMake, that I think has more capabilities. But I think that is out of scope for now.

Reply:
Honestly, It's been a while since I did that too so it was a suggestion. Deferring that to later PRs makes sense.